pylint/extensions/dict_init_mutate.py

```diff
@@ -0,0 +1,61 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/main/LICENSE
+# Copyright (c) https://github.com/PyCQA/pylint/blob/main/CONTRIBUTORS.txt
+
+"""Check for use of dictionary mutation after initialization."""
+
+from astroid import nodes
+
+from pylint.checkers import BaseChecker
+from pylint.checkers.utils import only_required_for_messages
+from pylint.interfaces import HIGH
+from pylint.lint.pylinter import PyLinter
+
+
+class DictInitMutateChecker(BaseChecker):
+    name = "dict-init-mutate"
+    msgs = {
+        "W3301": (
+            "Dictionary mutated immediately after initialization",
+            "dict-init-mutate",
+            "Dictionaries can be initialized with a single statement"
+            "using dictionary literal syntax.",
+        )
+    }
+
+    @only_required_for_messages("dict-init-mutate")
+    def visit_assign(self, node: nodes.Assign) -> None:
+        """
+        Detect dictionary mutation immediately after initialization.
+
+        At this time, detecting nested mutation is not supported.
+        """
+        if not isinstance(node.value, nodes.Dict):
+            return
+
+        dict_name = node.targets[0]
+        if len(node.targets) != 1 or not isinstance(dict_name, nodes.AssignName):
```

Comment:
It's annoying that we can't use the walrus operator here, because then these early returns could be fully streamlined.

Reply:
Indeed. https://endoflife.date/python 7 months